nuxeo-ai-similar-content/src/main/java/org/nuxeo/ai/similar/content/services/SimilarServiceComponent.java

```diff
@@ -84,6 +113,91 @@ public String getOperationID() {
         return operationID;
     }
 
+    @Override
+    public String index(String query, String user, boolean reindex) {
+        BulkServiceImpl bs = (BulkServiceImpl) Framework.getService(BulkService.class);
+        KeyValueStoreProvider kv = (KeyValueStoreProvider) bs.getKvStore();
+        boolean alreadyRunning = kv.keyStream(STATUS_PREFIX)
+                                   .map(kv::get)
+                                   .map(BulkCodecs.getStatusCodec()::decode)
+                                   .filter(status -> INDEX_ACTION_NAME.equals(status.getAction()))
+                                   .anyMatch(status -> status.getState().equals(BulkStatus.State.RUNNING)
+                                           || status.getState().equals(BulkStatus.State.SCHEDULED));
+        if (alreadyRunning) {
+            throw new ConcurrentUpdateException("Only one deduplication index action is allowed at a time");
+        }
+
+        if (!reindex) {
+            query += " AND ecm:mixinType != " + NXQL.escapeString(DEDUPLICATION_FACET);
```

Comment:
Should we not have this check on the query for the facet filter for all use cases?

Reply:
we can indeed check if such expression already takes place and apply it optionally. If initial query wasn't properly structured and contains this condition - it's a considered as a human error IMO